product_packaging_dimension/models/product_packaging.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2019-2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class ProductPackaging(models.Model):
+    _inherit = "product.packaging"
+    # Constraints and dimensional field names are identical
+    # to product_packaging.py in delivery.
+    # The redundancy here avoids unnecessary dependencies on sale modules.
+
+    _sql_constraints = [
+        ("positive_height", "CHECK(height>=0)", "Height must be positive"),
+        ("positive_width", "CHECK(width>=0)", "Width must be positive"),
+        ("positive_length", "CHECK(packaging_length>=0)", "Length must be positive"),
+        (
+            "positive_weight",
+            "CHECK(weight>=0)",
+            "Weight must be positive",
+        ),
+    ]
+    height = fields.Integer()
+    width = fields.Integer()
+    packaging_length = fields.Integer(string="Length")
+
+    length_uom_id = fields.Many2one(
+        "uom.uom",
+        "Dimensions Units of Measure",
+        domain=lambda self: [
+            ("category_id", "=", self.env.ref("uom.uom_categ_length").id)
+        ],
+        help="UoM for packaging length, height, width (based on lenght UoM)",
+        default=lambda self: self.env[
+            "product.template"
+        ]._get_length_uom_id_from_ir_config_parameter(),
+    )
+    length_uom_name = fields.Char(
+        string="Length unit of measure label",
+        related="length_uom_id.name",
+        readonly=True,
+    )
+
+    weight = fields.Float()
+    weight_uom_id = fields.Many2one(
+        "uom.uom",
+        string="Weight Units of Measure",
+        domain=lambda self: [
+            ("category_id", "=", self.env.ref("uom.product_uom_categ_kgm").id)
+        ],
+        help="Weight Unit of Measure",
+        compute=False,
+        default=lambda self: self.env[
+            "product.template"
+        ]._get_weight_uom_id_from_ir_config_parameter(),
+    )
+
+    weight_uom_name = fields.Char(
+        string="Weight unit of measure label",
+        related="weight_uom_id.name",
+        readonly=True,
+    )
+
+    volume = fields.Float(
```

Comment:
```suggestion
    volume = fields.Float(
        digits=(8, 4),
        compute="_compute_volume",
        help="The Packaging volume",
    )
```

> computed fields are not stored by default, they are computed and returned when requested.
> Computed fields are readonly by default. To allow setting values on a computed field, use the inverse parameter.
detail in [Computed Fields](https://www.odoo.com/documentation/18.0/developer/reference/backend/orm.html#computed-fields)